brownie/network/multicall.py

```diff
@@ -66,13 +68,14 @@ def _flush(self, future_result: Result = None) -> Any:
             # either all calls have already been made
             # or this result has already been retrieved
             return future_result
-        ContractCall.__call__.__code__ = getattr(ContractCall, "__original_call_code")
-        results = self._contract.tryAggregate(  # type: ignore
-            False,
-            [_call.calldata for _call in self._pending_calls],
-            block_identifier=self.block_number,
-        )
-        ContractCall.__call__.__code__ = getattr(ContractCall, "__proxy_call_code")
+        with self.lock:
```

Comment:
I'd place this lock at line 67 instead..  this way if one thread is blocked while another thread clears `_pending_calls`, the 2nd thread then doesn't try to do anything once it gets the lock.